design/src/rfcs/rfc0017_event_streams_errors.md

```diff
@@ -0,0 +1,183 @@
+<!-- Give your RFC a descriptive name saying what it would accomplish or what feature it defines -->
+RFC: Errors for event streams
```

Comment:
Some of the changes proposed here need some breaking changelog entries for the SDK.